### PR TITLE
add config option for other goodies

### LINKED ISSUE #1081: config for non shirt goodies
for events where there will be a different goodie than a shirt (for example the GPN) it would be usefull to have a "goodie score" instead of the "shirt score" and no personal shirt size setting

### PR DESCRIPTION
**Pull request by xuwhite**
fixes #1081

### CODE REVIEW

File: config/config.default.php [hunk 1]
```diff
@@ -276,6 +276,9 @@
     // Enables the T-Shirt configuration on signup and profile
     'enable_tshirt_size'      => (bool) env('ENABLE_TSHIRT_SIZE', true),
 
+    // When true changes everything from shirts to goodies and disables shirt size
+    'other_goodie'            => (bool) env('OTHER_GOODIE', true),
```
Comment (MyIgel) [inline config/config.default.php:280]: Having T-Shirts enabled by default would make more sense?
```suggestion
    'other_goodie'            => (bool) env('OTHER_GOODIE', false),
```

File: resources/lang/de_DE/default.po [hunk 2]
```diff
@@ -3150,3 +3150,34 @@ msgstr "Möchtest du den Arbeitseinsatz von %s wirklich löschen?"
 
 msgid "Number of shifts: %s"
 msgstr "Anzahl an Schichten: %s"
+
```
Comment (MyIgel) [inline resources/lang/de_DE/default.po:3153]: I would highly appreciate it if you could move those above the translations with dot notation to be less confusing (and also move the `Number of shifts: %s` up too when you are on it ;))

**Review by MyIgel — CHANGES_REQUESTED** [review]
Some first testing results:

It might be a good idea to generally refer to them as Goodies and just optionally add the T-Shirt info where necessary?

On the user page the `User state` / `User_view_state_admin` still shows `T-Shirt` instead of the expected `Goodie`.

File: includes/view/User_view.php [hunk 3]
```diff
@@ -401,7 +407,7 @@ function User_view_myshifts(
         ];
         if (config('enable_tshirt_size', false) && ($its_me || $tshirt_admin)) {
             $myshifts_table[] = [
-                'date'       => '<b>' . __('Your t-shirt score') . '&trade;:</b>',
+                'date'       => '<b>' . config('other_goodie') ? __('Your goodie score') : __('Your t-shirt score') . '&trade;:</b>',
```
[the remark this answers is not in the document]
Comment (xuwhite) [reply includes/view/User_view.php:410]: done

File: config/config.default.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (xuwhite) [reply config/config.default.php:280]: done

File: includes/view/User_view.php [hunk 4]
```diff
@@ -141,9 +143,13 @@ function Users_view(
...
     if (config('enable_tshirt_size')) {
-        $user_table_headers['got_shirt'] = Users_table_header_link('got_shirt', __('T-Shirt'), $order_by);
+        if (config('other_goodie')) {
+            $user_table_headers['got_shirt'] = Users_table_header_link('got_shirt', __('Goodie'), $order_by);
+        } else {
+            $user_table_headers['got_shirt'] = Users_table_header_link('got_shirt', __('T-Shirt'), $order_by);
+        }
     }
-    if (config('enable_tshirt_size')) {
+    if (config('enable_tshirt_size' && !config('other_goodie'))) {
```
[the remark this answers is not in the document]
Comment (xuwhite) [reply includes/view/User_view.php:152]: done

File: resources/lang/de_DE/default.po (hunk 2 above)
Comment (xuwhite) [reply resources/lang/de_DE/default.po:3153]: done

File: resources/lang/de_DE/default.po [hunk 5]
```diff
@@ -3150,3 +3150,34 @@ msgstr "Möchtest du den Arbeitseinsatz von %s wirklich löschen?"
...
+msgstr "Ausgegebene Goodies"
+
+msgid "Goodie statistic"
+msgstr "Goodie Statistik"
+
+msgid "remove goodie"
+msgstr "entferne Goodie"
+
+msgid "got goodie"
+msgstr "hat Goodie"
```
[the remark this answers is not in the document]
Comment (xuwhite) [reply resources/lang/de_DE/default.po:3174]: done
i changed the shirt equal too

File: includes/view/User_view.php [hunk 6]
```diff
@@ -532,7 +538,7 @@ function User_view(
             div('row', [
                 div('col-md-12', [
                     buttons([
-                        $auth->can('user.edit.shirt') && config('enable_tshirt_size') ? button(
+                        $auth->can('user.edit.shirt') && (config('enable_tshirt_size') && !config('other_goodie')) ? button(
```
[the remark this answers is not in the document]
Comment (xuwhite) [reply includes/view/User_view.php]: done

**Pull request merged**